functions/event-reminders-hourly/index.js

```diff
@@ -89,90 +89,100 @@ const handler = async function (event, context) {
+              blocks: [
+                {
+                  type: 'header',
+                  text: {
+                    type: 'plain_text',
+                    text: '⏰ Starting Soon:',
+                    emoji: true,
+                  },
+                },
+              ],
             };
-          }
 
-          message.blocks.push(titleBlock);
-
-          if (
-            event.eventJoinLink &&
-            event.eventJoinLink.substring(0, 4) !== 'http'
-          ) {
-            message.blocks.push({
+            const titleBlock = {
               type: 'section',
               text: {
                 type: 'mrkdwn',
-                text: `*Location:* ${event.eventJoinLink}`,
+                text: `*${
+                  event.title
+                }*\n<!date^${eventDate.toSeconds()}^{date_long_pretty} {time}|${eventDate.toFormat(
+                  'EEEE, fff'
+                )}>`,
               },
-            });
-          }
+            };
 
-          message.blocks.push(
-            {
-              type: 'context',
-              elements: [
-                {
+            if (
+              event.eventJoinLink &&
+              event.eventJoinLink.substring(0, 4) === 'http'
+            ) {
+              titleBlock.accessory = {
+                type: 'button',
+                text: {
+                  type: 'plain_text',
+                  text: 'Join Event',
+                  emoji: true,
+                },
+                value: `join_event_${event.id}`,
+                url: event.eventJoinLink,
+                action_id: 'button-join-event',
+              };
+            }
+
+            message.blocks.push(titleBlock);
+
+            if (
+              event.eventJoinLink &&
+              event.eventJoinLink.substring(0, 4) !== 'http'
+            ) {
+              message.blocks.push({
+                type: 'section',
+                text: {
                   type: 'mrkdwn',
-                  text: slackify(event.eventCalendarDescription),
+                  text: `*Location:* ${event.eventJoinLink}`,
                 },
-              ],
-            },
-            {
-              type: 'divider',
+              });
             }
-          );
 
-          return message;
+            message.blocks.push(
+              {
+                type: 'context',
+                elements: [
+                  {
+                    type: 'mrkdwn',
+                    text: slackify(event.eventCalendarDescription),
+                  },
+                ],
+              },
+              {
+                type: 'divider',
+              }
+            );
+
+            return message;
+          };
+
+          const messages = [createMessage(DEFAULT_SLACK_EVENT_CHANNEL)];
+
+          if (event.eventSlackAnnouncementsChannelId) {
+            messages.push(
+              createMessage(event.eventSlackAnnouncementsChannelId)
+            );
+          }
```

Comment:
Two things here:

1. Adjust to make sure `event.eventSlackAnnouncementsChannelId` !== `DEFAULT_SLACK_EVENT_CHANNEL` before adding
1. Let's abstract this in to one `announcementChannels` const at the top of this process (maybe just inside the `try {}` inside of `handler`). Then here we can do `channels.map(channel=>createMessage(channel)` and don't have to do the `if (event.eventSlackAnnouncementsChannelId)` again other places.

Reply:
How about this:

At top inside of first `try`:

```ts
const announcementsChannels = [DEFAULT_SLACK_EVENT_CHANNEL];

if (event.eventSlackAnnouncementsChannelId) {
  announcementsChannels.push(event.eventSlackAnnouncementsChannelId);
}
```

inside of the `flatMap` call

```diff
- const messages = [createMessage(DEFAULT_SLACK_EVENT_CHANNEL)];
- 
- if (event.eventSlackAnnouncementsChannelId) {
-   messages.push(
-     createMessage(event.eventSlackAnnouncementsChannelId)
-   );
- }
+ const messages = announcementsChannels.map(createMessage);
```

Inside of the `hourlyAdminMessage`:

```diff
text: {
  type: 'mrkdwn',
  text:
    `*Announcement posted to:* ` +
-     channels.map((channel) => `<#${channel}>`).join(' '),
+     announcementsChannels.map((channel) => `<#${channel}>`).join(' '),
},
```

Reply:
By "at top inside the first `try`" I believe you mean inside the `try` but outside of the `flatMap`. However, that won't work because `event` isn't defined yet (I think you may be mixing up the `event` that is passed into the `handler` and the `event` variable that gets defined in the `flatMap`?

Also, how did you post those code blocks with the green and red diffs?

Reply:
Also, how did you post those code blocks with the green and red diffs?

Reply:
> I think you may be mixing up the `event` that is passed into the `handler()` and the `event` variable that gets defined inside the `flatMap`

...shit you're right 😂

Let's rename one of those while you're in there so I don't make that mistake again.

> Also, how did you format those code blocks with the green and red diffs?

Use the `diff` "language" in the code fence, and add `- ` and `+ ` where needed:

```diff
Here is some code
- I'm removing this line
+ And adding this line
```

````
```diff
Here is some code
- I'm removing this line
+ And adding this line
```
````

Reply:
> 1. Adjust to make sure `event.eventSlackAnnouncementsChannelId` !== `DEFAULT_SLACK_EVENT_CHANNEL` before adding

done 👍

Reply:
> Let's rename one of those while you're in there so I don't make that mistake again.

done 👍